### PR TITLE
Fix warnings for deprecated apis with kube 1.16+

### LINKED ISSUE #1591: k8s 1.16+ deprecation warnings in development profile
/kind bug

* apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+
* admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+
* using legacy gatewayName format "knative-ingress-gateway.knative-serving"
* admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+

**What steps did you take and what happened:**
`kustomize build config/overlays/development | kubectl apply --validate=false -f -`
```
namespace/kfserving-system created
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
customresourcedefinition.apiextensions.k8s.io/inferenceservices.serving.kubeflow.org created
customresourcedefinition.apiextensions.k8s.io/trainedmodels.serving.kubeflow.org created
Warning: admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 MutatingWebhookConfiguration
mutatingwebhookconfiguration.admissionregistration.k8s.io/inferenceservice.serving.kubeflow.org created
serviceaccount/kfserving-models-web-app created
role.rbac.authorization.k8s.io/leader-election-role created
clusterrole.rbac.authorization.k8s.io/kfserving-manager-role created
clusterrole.rbac.authorization.k8s.io/kfserving-models-web-app-cluster-role created
clusterrole.rbac.authorization.k8s.io/kfserving-proxy-role created
rolebinding.rbac.authorization.k8s.io/leader-election-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/kfserving-manager-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/kfserving-models-web-app-binding created
clusterrolebinding.rbac.authorization.k8s.io/kfserving-proxy-rolebinding created
configmap/inferenceservice-config created
configmap/kfserving-config created
configmap/kfserving-models-web-app-config-6t9ktmdt9c created
secret/kfserving-webhook-server-secret created
service/kfserving-controller-manager-metrics-service created
service/kfserving-controller-manager-service created
service/kfserving-models-web-app created
service/kfserving-webhook-server-service created
deployment.apps/kfserving-models-web-app created
statefulset.apps/kfserving-controller-manager created
certificate.cert-manager.io/serving-cert created
issuer.cert-manager.io/selfsigned-issuer created
Warning: using legacy gatewayName format "knative-ingress-gateway.knative-serving"; prefer the <namespace>/<name> format: "knative-serving/knative-ingress-gateway"
virtualservice.networking.istio.io/kfserving-models-web-app created
Warning: admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 ValidatingWebhookConfiguration
validatingwebhookconfiguration.admissionregistration.k8s.io/inferenceservice.serving.kubeflow.org created
validatingwebhookconfiguration.admissionregistration.k8s.io/trainedmodel.serving.kubeflow.org created
```

**What did you expect to happen:**
no warnings

**Environment:**

- Istio Version: 1.8.2
- Knative Version: 0.22
- KFServing Version: master branch
- Kubeflow version: -
- Kubernetes version: (use `kubectl version`): 1.19

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1591 

**Special notes for your reviewer**:

1. Fix deprecated api group for validating/mutating webhooks
2. Fix deprecated format for istio gateway resource

**Release note**:

```release-note

```

### CODE REVIEW

File: config/default/webhookcainjection_patch.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ metadata:
 webhooks:
   - name: inferenceservice.kfserving-webhook-server.defaulter
```
Comment (theofpa) [inline config/default/webhookcainjection_patch.yaml:10]:
```suggestion
  - name: inferenceservice.kfserving-webhook-server.defaulter
    clientConfig:
    service:
      name: $(webhookServiceName)
```

File: config/default/webhookcainjection_patch.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ metadata:
 webhooks:
   - name: inferenceservice.kfserving-webhook-server.validator
```
Comment (theofpa) [inline config/default/webhookcainjection_patch.yaml:19]:
```suggestion
  - name: inferenceservice.kfserving-webhook-server.validator
    clientConfig:
    service:
      name: $(webhookServiceName)
```

**Review by theofpa — COMMENTED** [review]
for some reason the webhookservice name is not propagated by kustomize to the webhook definition, so adding it in all webhooks will fix the error of `make deploy-ci`. I can't suggest the `trainedmodel` change, but it should be the same as in `inferenceservice`'s patch.

**Comment by yuzisun** [comment]
> for some reason the webhookservice name is not propagated by kustomize to the webhook definition, so adding it in all webhooks will fix the error of `make deploy-ci`. I can't suggest the `trainedmodel` change, but it should be the same as in `inferenceservice`'s patch.

I think the kustomize version might be too low in the test worker image, it works locally for me.

**Comment by yuzisun** [comment]
failed on 
```
{"level":"error","ts":1621644612.4082367,"logger":"controller-runtime.manager.controller.inferenceservice","msg":"Reconciler error","reconc
iler group":"serving.kubeflow.org","reconciler kind":"InferenceService","name":"isvc-triton","namespace":"kfserving-ci-e2e-test","error":"I
nternal error occurred: failed calling webhook \"inferenceservice.kfserving-webhook-server.defaulter\": webhook returned response.patchType
 but not response.patch","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.
0/pkg/internal/controller/controller.go:267\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/g
o/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:235\nsigs.k8s.io/controller-runtime/pkg/internal/cont
roller.(*Controller).Start.func1.1\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.0/pkg/internal/controller/controller.go:198\nk8s.io/a
pimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:185\nk8s.io/apimach
inery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/w
ait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go
/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\t/go/pkg/mod/k8s
.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\t/go/pkg/mod/k8s.io/apimachinery@v
0.19.2/pkg/util/wait/wait.go:99"}
2021/05/22 00:50:40 http: panic serving 192.168.110.253:49738: runtime error: invalid memory address or nil pointer dereference
goroutine 50517 [running]:
```

File: hack/quick_install.sh [hunk 3]
```diff
@@ -1,8 +1,8 @@
 set -e
 
 export ISTIO_VERSION=1.9.0
-export KNATIVE_VERSION=v0.18.0
-export KFSERVING_VERSION=v0.5.1
+export KNATIVE_VERSION=v0.22.0
+export KFSERVING_VERSION=v0.6.0-rc0
```
Comment (NikeNano) [inline hack/quick_install.sh:5]: Should this be a stable version?

File: test/scripts/run-e2e-tests.sh [hunk 4]
```diff
@@ -32,6 +32,13 @@ echo "Upgrading kubectl ..."
 wget -q -O /usr/local/bin/kubectl https://dl.k8s.io/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl
 chmod a+x /usr/local/bin/kubectl
 
+echo "Installing kustomize ..."
+which kustomize
```
Comment (NikeNano) [inline test/scripts/run-e2e-tests.sh:36]: What is the purpose of `which` here?

**Pull request closed without merging**